File: 1.两数之和.js
/*
 * @lc app=leetcode.cn id=1 lang=javascript
 *
 * [1] 两数之和
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const hash = {}
    for (let i = 0; i < nums.length; i++) {
        const val = nums[i]
        if (hash[target - val] !== undefined) {
            return [hash[target - val], i]
        }
        if (hash[val] === undefined) {
            hash[val] = i;
        }
    }
    return [];
};
// @lc code=end

File: 1.两数之和.rs
/*
 * @lc app=leetcode.cn id=1 lang=rust
 *
 * [1] 两数之和
 */

// @lc code=start
impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut map: HashMap<i32, usize> = HashMap::new();
        for (index, &value) in nums.iter().enumerate() {
            if let Some(&i) = map.get(&(target - value)) {
                return vec![i as i32, index as i32];
            }
            map.insert(value, index);
        }
        panic!()
    }
}
// @lc code=end
